### ISSUE TITLE
second ds.phase_correct(forceNew=True) fails

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#166]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/166) by Young Il Joe (Bitbucket: [jojo978](https://bitbucket.org/jojo978), GitHub: [jojo978](https://github.com/jojo978)).**

----------------------------------------

Somehow, the second \(and later one\) phase\_correct\(forceNew=True\) fails with following messages on my windows box.

```
RuntimeError Traceback (most recent call last)
<ipython-input-14-e56cc12ac55f> in <module>
----> 1 ds.phase_correct(forceNew=True)
c:\users\ynj5\virt_env\py374\lib\site-packages\mass-0.6.6-py3.7-win-amd64.egg\mass\core\channel.py in phase_correct(self, forceNew, category, ph_peaks, method2017, kernel_width, save_to_hd
f5)
    1583
    1584 if save_to_hdf5:
--> 1585     self.phaseCorrector.toHDF5(self.hdf5_group, overwrite=True)
    1586
    1587     LOG.info('Channel %3d phase corrected. Correction size: %.2f',
c:\users\ynj5\virt_env\py374\lib\site-packages\mass-0.6.6-py3.7-win-amd64.egg\mass\core\phase_correct.py in toHDF5(self, hdf5_group, name, overwrite)
    19
    20 def toHDF5(self, hdf5_group, name="phase_correction", overwrite=False):
--> 21     hdf5_group["{}/phase_uniformifier_x".format(name)] = self.phase_uniformifier_x
    22     hdf5_group["{}/phase_uniformifier_y".format(name)] = self.phase_uniformifier_y
    23     hdf5_group["{}/uncorrected_name".format(name)] = self.uncorrectedName
c:\users\ynj5\virt_env\py374\lib\site-packages\h5py_hl\group.py in setitem(self, name, obj)
    384 else:
    385     ds = self.create_dataset(None, data=obj, dtype=base.guess_dtype(obj))
--> 386     h5o.link(ds.id, self.id, name, lcpl=lcpl)
    387
    388 if do_link:
h5py_objects.pyx in h5py._objects.with_phil.wrapper()
h5py_objects.pyx in h5py._objects.with_phil.wrapper()
h5py\h5o.pyx in h5py.h5o.link()
RuntimeError: Unable to create link (name already exists)
‌
```

‌

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Can you make a test for this? I don’t think I understand it.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Now can run phase_correct twice

When running phase_correct on a dataset a second time (with `forceNew=True`), HDF5 library would fail to write the relevant data. Fixes #166.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Removing milestone: v0.7 (automated comment)
